From f3b3ec8c13dd83393867e40e429fdc7b6b49663e Mon Sep 17 00:00:00 2001
From: Rishi Gupta <rishgupt@codeaurora.org>
Date: Mon, 1 Mar 2021 15:48:57 +0530
Subject: [PATCH] ima-evm-utils: add support for verifying portable signature

Portable signature for EVM starts with 05. Current code
was not checking for this. This commit adds that check
so that verification of portable EVM signature can be
done.

Signed-off-by: Rishi Gupta <rishgupt@codeaurora.org>
---
 src/evmctl.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/evmctl.c b/src/evmctl.c
index 3d2a10b..e8e8fc6 100644
--- a/src/evmctl.c
+++ b/src/evmctl.c
@@ -824,8 +824,8 @@ static int verify_evm(const char *file)
 		return len;
 	}
 
-	if (sig[0] != 0x03) {
-		log_err("%s has no signature\n", xattr_evm);
+	if ((sig[0] != 0x03) && (sig[0] != 0x05)) {
+		log_err("%s has no signature, 0th byte=%x\n", xattr_evm, sig[0]);
 		return -1;
 	}
 
-- 
2.7.4

